homeassistant/components/comelit/coordinator.py

```diff
@@ -73,8 +71,9 @@ async def _async_update_data(self) -> dict[str, Any]:
         logged = False
         try:
             logged = await self.api.login()
-        except (asyncio.exceptions.TimeoutError, aiohttp.ClientConnectorError) as err:
+        except exceptions.CannotConnect as err:
             _LOGGER.warning("Connection error for %s", self._host)
+            await self.api.close()
             raise UpdateFailed(f"Error fetching data: {repr(err)}") from err
         finally:
             if not logged:
```

Comment:
I missed this here. When CannotConnect raised, it will trigger a UpdateFailed, which in turn will trigger ConfigEntryAuthFailed. Seems weird, is that expected?

Reply:
Addressed in https://github.com/home-assistant/core/pull/102768